ublox/src/ubx_packets/packets.rs

```diff
@@ -2404,6 +2462,15 @@ impl<'a> RxmRawxInfoIter<'a> {
     }
 }
 
+impl serde::Serialize for RxmRawxInfoIter<'_> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.collect_seq(self.clone())
```

Comment:
can just clone or something... was playing around with refcell couldn't get that to work but cloning seems the simpliest

Reply:
Cloning the iterator is fine, there is just one thing to watch out for when cloning an iterator. When you clone you generally are not cloning the underlying data, but rather the state of the iterator. Like if you have a

```rust
let slice = &[1,2,3,4,5];
let mut it = slice.chunks_exact(2);
it.next(); // &[1,2]
let mut it2 = it.clone();
it.next(); // &[3,4]
it2.next(); // &[3,4]
```

when you clone the partially used `it`, `it2` will be at the same place (rather than a new iterator starting from the beginning of the slice. 

So, if you attempted to serialize a partially used `RxmRawxInfoIter`, you'd only serialize the elements that haven't been consumed yet.

In this case I'm not even sure you need to impl serialize for `RxmRawxInfoIter` (or any of the iterators really). Instead you can just handle the iterator in the Serialize impl for whatever message holds the iterator

Reply:
The only one that probably needs serialize impl is the `U32Iter`, because it doesn't belong to a packet it comes from an iterator on  `EsfMeasInfo`. That way you can derive Serialize on EsfMeasInfo

Reply:
don't think it should ever gets consumed tho,
the `from` function only gets called whenever we do for example
`EsfRawRef#iter()` which runs `EsfRawIter::new(val)` every time 
and when we serialize, we run like `&self.iter()` which would create a new iterator always fresh